va/validation-authority.go

```diff
@@ -333,6 +333,16 @@ func (va *ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdenti
 		return challenge, err
 	}
 
+	contentTypes, ok := httpResponse.Header[http.CanonicalHeaderKey("content-type")]
+	if ok && (len(contentTypes) != 1 || (len(contentTypes) == 1 && contentTypes[0] != "application/jose+json")) {
```

Comment:
I would split this into two problem cases:

```
if ok && len(contentTypes) != 1 {
  ...
  Detail: "More than one Content-Type header provided"
} else if ok && contentTypes[0] != "application/jose+json" {
  ...
  Detail: "Incorrect Content-Type header. Must be application/jose+json or absent"
```

Also please make sure to test each case: header absent, header present and equal to application/jose+json, header present but not equal to application/jose+json, multiple headers present.